web/api/v1/api.go

```diff
@@ -506,6 +525,94 @@ func (api *API) alertmanagers(r *http.Request) (interface{}, *apiError) {
 	return ams, nil
 }
 
+// AlertDiscovery has info for all alerts
+type AlertDiscovery struct {
+	Alertgrps []*Alertgrp `json:"alertgrp"`
+}
+
+// Alert has info for a alert
+type Alert struct {
+	Labels      labels.Labels `json:"labels"`
+	Status      string        `json:"status"`
+	Activesince *time.Time    `json:"activesince,omitempty"`
+}
+
+// Alertgrp has info for alerts part of a group
+type Alertgrp struct {
+	Name        string        `json:"name"`
+	Query       string        `json:"query"`
+	Duration    string        `json:"duration"`
+	Annotations labels.Labels `json:"annotations,omitempty"`
+	Alerts      []*Alert      `json:"alerts"`
+}
+
+func (api *API) alerts(r *http.Request) (interface{}, *apiError) {
+	alertingrules := api.alertsrulesRetreiver.AlertingRules()
+	var alertgrps []*Alertgrp
+	res := &AlertDiscovery{Alertgrps: alertgrps}
+	for _, activerule := range alertingrules {
+		t := &Alertgrp{
+			Name:        activerule.Name(),
+			Query:       fmt.Sprintf("%v", activerule.Query()),
+			Duration:    activerule.Duration().String(),
+			Annotations: activerule.Annotations(),
+		}
+		alerts := activerule.Alertinfo()
+		var activealerts []*Alert
+		for _, alert := range alerts {
+			q := &Alert{
+				Labels:      alert.Labels,
```

Comment:
Do we have the annotations too?

Reply:
yes in the alerts api see below:
output of alerts api
```
curl -s localhost:9090/api/v1/alerts | jq '.'
{
  "status": "success",
  "data": {
    "alertgrp": [
      {
        "name": "R1file1",
        "query": "up{job=\"prometheus\"} == 1",
        "duration": "2m0s",
        "annotations": {
          "dashboard": "http://grafana.com/somelink",
          "description": "rule1 in file 1",
          "summary": "rule1 in file 1"
        },
        "alerts": [
          {
            "labels": {
              "alertname": "R1file1",
              "env": "prod",
              "instance": "localhost:9090",
              "job": "prometheus",
              "severity": "high"
            },
            "status": "pending",
            "activesince": "2018-03-26T17:48:03.231816564-07:00"
          }
        ]
      },
      {
        "name": "R2file1",
        "query": "up{job=\"prom\"} == 0",
        "duration": "2m0s",
        "annotations": {
          "dashboard": "http://grafana.com/somelink",
          "description": "rule2 in file 1",
          "summary": "rule2 in file 1"
        },
        "alerts": [
          {
            "labels": {
              "alertname": "R2file1",
              "env": "prod",
              "instance": "localhost:9091",
              "job": "prom",
              "severity": "critical"
            },
            "status": "pending",
            "activesince": "2018-03-26T17:48:03.231816564-07:00"
          }
        ]
      },
      {
        "name": "R1file2",
        "query": "up != 2",
        "duration": "10s",
        "annotations": {
          "dashboard": "http://grafana.com/somelink",
          "description": "rule1 in file 2",
          "summary": "rule1 in file 2"
        },
        "alerts": [
          {
            "labels": {
              "alertname": "R1file2",
              "env": "test",
              "instance": "localhost:9090",
              "job": "prometheus",
              "severity": "high"
            },
            "status": "pending",
            "activesince": "2018-03-26T17:48:05.139489603-07:00"
          },
          {
            "labels": {
              "alertname": "R1file2",
              "env": "test",
              "instance": "localhost:9091",
              "job": "prom",
              "severity": "high"
            },
            "status": "pending",
            "activesince": "2018-03-26T17:48:05.139489603-07:00"
          }
        ]
      },
      {
        "name": "HighErrors",
        "query": "(100 - (avg by(instance) (irate(node_cpu{mode=\"idle\",name=\"node-exporter\"}[5m])) * 100)) > 75",
        "duration": "5m0s",
        "annotations": {
          "description": "Instance:CPU usage is above 75% "
        },
        "alerts": null
      }
    ]
  }
}
```

output of rules api:
```
curl -s localhost:9090/api/v1/rules | jq '.'
{
  "status": "success",
  "data": {
    "groups": [
      {
        "name": "rule_grp1",
        "file": "/home/gardame/allrules/grp1_rules.yml",
        "rules": [
          {
            "rule": "alert: R1file1\nexpr: up{job=\"prometheus\"} == 1\nfor: 2m\nlabels:\n  env: prod\n  severity: high\nannotations:\n  dashboard: http://grafana.com/somelink\n  description: rule1 in file 1\n  summary: rule1 in file 1\n"
          },
          {
            "rule": "alert: R2file1\nexpr: up{job=\"prom\"} == 0\nfor: 2m\nlabels:\n  env: prod\n  severity: critical\nannotations:\n  dashboard: http://grafana.com/somelink\n  description: rule2 in file 1\n  summary: rule2 in file 1\n"
          }
        ]
      },
      {
        "name": "rule_grp2",
        "file": "/home/gardame/allrules/grp2_rules.yml",
        "rules": [
          {
            "rule": "alert: R1file2\nexpr: up != 2\nfor: 10s\nlabels:\n  env: test\n  severity: high\nannotations:\n  dashboard: http://grafana.com/somelink\n  description: rule1 in file 2\n  summary: rule1 in file 2\n"
          }
        ]
      },
      {
        "name": "grp3",
        "file": "/home/gardame/allrules/grp3_rules.yml",
        "rules": [
          {
            "rule": "record: instance:errors:rate5m\nexpr: rate(errors_total[5m])\n"
          },
          {
            "rule": "record: instance:requests:rate5m\nexpr: rate(requests_total[5m])\n"
          },
          {
            "rule": "alert: HighErrors\nexpr: (100 - (avg by(instance) (irate(node_cpu{mode=\"idle\",name=\"node-exporter\"}[5m]))\n  * 100)) > 75\nfor: 5m\nlabels:\n  severity: critical\nannotations:\n  description: 'Instance:CPU usage is above 75% '\n"
          }
        ]
      }
    ]
  }
}
```

Reply:
I think @brian-brazil is referring to the annotations in the active alerts (eg after being expanded by the templating engine).